### ISSUE TITLE
semconv:  rpc.type indicate server or client?

### ISSUE THREAD

**Issue by 1046102779** [open-telemetry/opentelemetry-specification#2419]
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#attributes

How do we use semantic convention:   `rpc`  to represent  server and client?

**Comment by arminru** [comment]
Hi @1046102779!
The OpenTelemetry Tracing API already defines dedicated properties for both a "Span Kind" and a "Status" on Spans.
Take a look here to learn more:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status

**Comment by 1046102779** [comment]
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#attributes

But how does metrics use tracing's `SpanKind` and `status`?

**Comment by arminru** [comment]
I misunderstood that this was about metrics, not tracing - sorry!
In this case it could make sense to introduce generic attributes for both a ~span~ "party/counterpart" kind (likely with the same values as defined for SpanKind) and an operation status so they can also be used for HTTP, FaaS, messaging and others.
What do you think? Would you like to drive spec PRs for these?

**Comment by 1046102779** [comment]
> In this case it could make sense to introduce generic attributes for both a ~span~ "party/counterpart" kind (likely with the same values as defined for SpanKind) and an operation status so they can also be used for HTTP, FaaS, messaging and others.
> What do you think? Would you like to drive spec PRs for these?

Yes. I think it's a general concept and it's appropriate to add to the opentelemetry specification, and I'm willing to drive this PR.

**Comment by arminru** [comment]
@1046102779 I just looked over the semantic conventions you were referring to again. In your screenshot, only the common metric **attributes** are shown. The metrics (i.e., the **metric names**) themselves already distinguish between client- and server-side observations (e.g., `rpc.server.duration` vs `rpc.client.duration`), however.
Because of that, there is no need for any attribute to distinguish clients from servers, right?

If additional RPC metrics were introduced, they would stick to this naming scheme with server/client in the namespace and this would also be the only way to have disjoint observations. Otherwise you would only be able to get meaningful results when aggregating metrics if you filter or split by the respective attribute.

The **status**, on the other hand, is indeed missing and I think it would make sense to introduce a generic metric attribute convention for this that matches the definitions for [Span Status](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status=). This attribute should be re-usable across all technologies/frameworks and not just be defined for RPC alone.

cc @trask @jsuereth @jmacd @reyang @pirgeo for your opinions

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Purely a semantic-conventions ask; the referenced spec PR #2430 ("add rpc.type and rpc.status for metrics") was closed unmerged. Maintainer @arminru clarified rpc.type is unnecessary (client/server split by metric names like rpc.server.duration) and rpc.status should be a generic attribute; semconv development has since moved to open-telemetry/semantic-conventions, where RPC conventions (incl. rpc.grpc.status_code) now live, so any remaining work belongs there.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Verified: pure semconv ask; PR #2430 closed unmerged 2022-04-10, maintainer @arminru ruled rpc.type unneeded (client/server split by metric name) and status should be generic; semconv moved to open-telemetry/semantic-conventions where the status concept now exists as rpc.response.status_code, and issue has been dormant since 2022 -- moved-repo close is correct.

Claude evaluated this issue with medium confidence that it can be closed.

Claude has classified this issue as belonging in another repository.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.
